### PR TITLE
feat(wrangler): Add support for `wrangler.jsonc`

### LINKED ISSUE #5437: 🚀 Feature Request: `wrangler.json` supports comments but wrangler won't pick up `wrangler.jsonc`
### Describe the solution

I find comments in the wrangler config file to be useful and I'm happy that they're supported in the json format too. But every ide complains about comments in json files unless it's `*.jsonc`, but it looks like wrangler with `-j` only looks for `wrangler.json` not `wrangler.jsonc` too.

Could we have wrangler check for either to be present?

### PR DESCRIPTION
**Pull request by CarmenPopoviciu**
## What this PR solves / how to test

This PR adds support for `wrangler.jsonc` config file, behind the same `--experimental-json-config` flag as `wrangler.json` support.

Fixes #5437

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [x] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [ ] Not necessary because:

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
LGTM!

**Comment by petebacondarwin** [comment]
I note you have said that a docs update is required before merge. Is that correct? Is the JSON support actually documented, given that it is experimental?

**Comment by CarmenPopoviciu** [comment]
I was thinking in https://developers.cloudflare.com/workers/wrangler/commands/ in the following section 👇 ?
Screenshot 2024-07-24 at 09 56 40

**Comment by petebacondarwin** [comment]
Ah I see. I guess that could be tweaked. But I won't block this PR on it.

**Pull request merged**
